### DISCUSSION TITLE
How to create a 'single page' module?

### DISCUSSION THREAD

**Discussion by DaniWinter** [area17/twill#1034] [Q&A]
I need a navigation item that leads to 1 form with 2 wysiwyg editors and an image uploader.
When I use the `php artisan twill:module` command it also generates an overview page I don't need.

**Comment by pboivin** [comment]
Hey @DaniWinter,

Checkout this previous discussion: https://github.com/area17/twill/issues/54#issuecomment-433988352

Especially point 2. from @ifox :

> One module per page, which means a single record in each associated table, and no listing in the CMS, but direct access to the form through the navigation, by using a route name in twill-navigation.php instead of 'module' => true. This is working great for say big landing pages that requires a lot of copy fields, browsers, repeaters and so on. [...]


Here's a quick example from a previous project :


In `twill-navigation.php` :

```
    'contact' => [
        'title' => 'Contact',
        'route' => 'admin.contact',
        'module' => false,
    ],
```


In `routes/admin.php` :
```
    Route::module('contactPages');
    Route::get('contact', 'ContactPageController@editPage')->name('contact');
```


In `app/Http/Controllers/Admin/ContactPageController.php` :
```
    public function editPage()
    {
        $model = "App\\Models\\{$this->getModelName()}";

        $page = App::make($model)->first();

        abort_unless($page, 500, "{$this->getModelName()} is missing");

        Session::put('pages_back_link', url()->current());

        return view("admin.{$this->moduleName}.form", $this->form($page->id));
    }
```

Hope this can help!
